### ISSUE TITLE
TensortRT Invalid data type: 'int32' when converting TF Object Detection graph

### ISSUE THREAD

**Issue by louisquinn** [tensorflow/tensorflow#17881]
### System information
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**: Yes see gist (change line 6 to object detection frozen inference graph path): https://gist.github.com/louisquinn/0c6729a32e87e899ece317de84d02acc
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: 16.04
- **TensorFlow installed from (source or binary)**: Source
- **TensorFlow version (use command below)**: 1.7
- **Python version**: 3.5
- **Bazel version (if compiling from source)**: 0.11.1
- **GCC/Compiler version (if compiling from source)**: 5.4.0
- **CUDA/cuDNN version**: CUDA 9.0 / cuDNN 7.1
- **GPU model and memory**: GTX 1080 8GB
- **Exact command to reproduce**: See gist (change line 6 to object detection frozen inference graph path): https://gist.github.com/louisquinn/0c6729a32e87e899ece317de84d02acc

### Describe the problem
Getting the following error when calling `trt.create_inference_graph` (see line 18 of gist):
`tensorflow/contrib/tensorrt/convert/convert_nodes.cc:2503] Non-OK-status: ConvertDType(tf_dtype, &dtype) status: Invalid argument: Unsupported data type int32`

Is int32 not supported by TensorRT, not yet implemented in the Tensorflow wrap, or am I doing something incorrect? This happens with all pre-trained TF Object Detection models.

**Comment by samikama** [comment]
@louisquinn at the time our error messages were not very user friendly. Can you tell which network are you trying? it is possible that it is not fully supported yet and error message could be misleading.

**Comment by chiachunfu** [comment]
Running into the same error message with custom SSD model.

**Comment by louisquinn** [comment]
@samikama 
You can reproduce this error using any of the **Object Detection** models from here:
https://github.com/tensorflow/models/blob/master/research/object_detection/g3doc/detection_model_zoo.md

I have not attempted TensorRT with the **Mask** models yet.

**Comment by louisquinn** [comment]
@samikama @petewarden 
Hi guys is there any update on this? Were you able to reproduce the error?

**Comment by chiachunfu** [comment]
@louisquinn you can try commenting out trt_shfn.cc line 52 under tensorflow/contrib/tensorrt/shape_fn and see if that works for you. Mine works fine after commenting out that line.

**Comment by samikama** [comment]
Sorry I couldn't get to that. I will try to get to it tomorrow. Meanwhile can you try with master to see if error message is more sensible. You can also try to increase minimum_segment_size to some large number. In general object detection network support will improve with TensorRT4.0 since 3.0.4 is missing some operators.

**Comment by louisquinn** [comment]
@samikama Okay great I will try your suggestions and update here.

**Comment by samikama** [comment]
@louisquinn I went through your test code with ssd_inception_v2_coco model from object detection zoo. I should have read the issue earlier :) Yes int32 ops are not available natively in TensorRT. It will be possible to convert object detection networks and others through plugin mechanism in near future with new TensorRT releases. Sorry for the inconvenience.

**Comment by samikama** [comment]
Closing issue. INT32 nodes are not supported in TensorRT until release 4.0

**Comment by louisquinn** [comment]
Okay great, thanks @samikama for the update.
